### PR TITLE
Add way of deploying pull requests to https://sr-org-scratch.github.io

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: .github/workflows/build-gh-pages.yml [hunk 1]
```diff
@@ -43,3 +43,32 @@ jobs:
...
           git commit -m "update with results of commit ${{ github.sha }}"
           git push -f origin gh-pages
+      - name: Deploy ssh key for scratch site (pull request)
+        if: ${{ github.event_name == 'pull_request' && github.ref_name != 'master' && github.repository == 'OSGeo/spatialreference.org' }}
+        shell: bash -l {0}
+        run: |
+          sudo apt install hxtools
+          mkdir $HOME/.ssh
+          # Note: this is obviously unsecure !!! But we don't mind if the content of sr-org-scratch/sr-org-scratch is "compromised"
+          cat .github/workflows/some_secret.txt | rot13 > $HOME/.ssh/id_rsa
```
Comment (jjimenezshaw) [inline .github/workflows/build-gh-pages.yml:53]: Is there any reason to not use a proper secret, as a few lines above?

File: .github/workflows/build-gh-pages.yml [hunk 2]
```diff
@@ -43,3 +43,32 @@ jobs:
...
+          chmod 700 $HOME/.ssh && chmod 600 $HOME/.ssh/id_rsa
+          ssh-keyscan -t rsa github.com >> $HOME/.ssh/known_hosts
+          eval `ssh-agent -s`
+          ssh-add $HOME/.ssh/id_rsa
+      - name: Deploy HTML pages to sratch site (pull request)
```
Comment (jjimenezshaw) [inline .github/workflows/build-gh-pages.yml:58]:
```suggestion
      - name: Deploy HTML pages to scratch site (pull request)
```

**Review by jjimenezshaw — COMMENTED** [review]
If I understood the PR correctly, it will "publish" the webpage at sr-org-scratch.github.io for any submitted pull request. Right? So the last submitted PR may stay in that url for a long time.
It would be nice to have a way to "clean" it.

File: .github/workflows/build-gh-pages.yml (hunk 1 above)
Comment (rouault) [reply .github/workflows/build-gh-pages.yml:53]: > Is there any reason to not use a proper secret, as a few lines above?

github secrets are only working on push, not pull requests

**Comment by rouault** [comment]
> Right? So the last submitted PR may stay in that url for a long time.
> It would be nice to have a way to "clean" it.

Is that a problem? I've added a robots.txt file so that the site isn't hopefully indexed

File: .github/workflows/build-gh-pages.yml (hunk 2 above)
Comment (rouault) [reply .github/workflows/build-gh-pages.yml]: done

**Comment by rouault** [comment]
I've added a commit that adds a clean-scratch-site action that can be manually run from the https://github.com/OSGeo/spatialreference.org/actions tab (once this PR is merged) and that resets the sr-org-scratch/sr-org-scratch.github.io repo

**Review by jjimenezshaw — APPROVED** [review]
You were fast @rouault 
I was going to suggest exactly what you did in clean-scratch-site.yml

**Comment by rouault** [comment]
https://sr-org-scratch.github.io/ now live with this PR

**Pull request merged**